### PR TITLE
Store all the violations in a centralized table.

### PR DESCRIPTION
**Pull request by blueandgold**
This is to address the main item in #894, by moving the violation tables from a per-model basis, and into a centralized table.  Will update tests if this PR's approach is okay.

Also, there is a secondary item to make it clear where the violations is coming from.  Do you have any suggestions?  I was thinking to use the ```full_name```, that has the full hierarchy path of the resource.  How do you think?

### CODE REVIEW

File: google/cloud/forseti/services/scanner/dao.py [hunk 1]
```diff
@@ -40,14 +40,17 @@ def define_violation(model_name, dbengine):
     """
 
     base = declarative_base()
-    violations_tablename = '{}_violations'.format(model_name)
+    # pylint: disable=too-many-format-args
```
Comment (ahoying) [inline google/cloud/forseti/services/scanner/dao.py]: Remove the pylint disable and remove the .format() from the string. The tablename is just 'violations'

File: google/cloud/forseti/services/scanner/dao.py [hunk 2]
```diff
@@ -40,14 +40,17 @@ def define_violation(model_name, dbengine):
...
+    # pylint: enable=too-many-format-args
 
     class Violation(base):
         """Row entry for a violation."""
 
         __tablename__ = violations_tablename
 
         id = Column(Integer, primary_key=True)
+        model_handle = Column(String(256))
         resource_type = Column(String(256), nullable=False)
```
Comment (ahoying) [inline google/cloud/forseti/services/scanner/dao.py]: Please add resource_id back into the model, this got dropped. Also I agree that adding resource_full_name is a good idea.

File: google/cloud/forseti/services/scanner/dao.py (hunk 1 above)
Comment (blueandgold) [reply google/cloud/forseti/services/scanner/dao.py]: Done.

File: google/cloud/forseti/services/scanner/dao.py (hunk 2 above)
Comment (blueandgold) [reply google/cloud/forseti/services/scanner/dao.py]: resource_id is added back; also taking the chance to fix how resource_id is being set that will make it more easy to identify which resource is the violation coming from.

I also started to implement the full_name, but it grew to be a sweeping change.  Let me do this in a follow-up PR, as I need to re-think how to make the easiest change.

**Pull request merged**